### DISCUSSION TITLE
How to get a `Date` object from `q-date`?

### DISCUSSION THREAD

**Discussion by Shayan-To** [quasarframework/quasar#12531] [General - Components / Directives / etc] [answered]
I have a `q-date` with `calendar="persian"`. The `modelValue` contains the localized date string, and my server is expecting the date in ISO format. The conversion utilities are not exported, and I couldn't find any way to get the `Date` object.

What can I do?

**Comment by pikil** [comment]
@Shayan-To, you can try to extract Date object with `extractDate()` method
https://quasar.dev/quasar-utils/date-utils#extract-date

**Reply by Shayan-To** [reply]
This method does not support date strings in Persian calendar.

**Reply by pikil** [reply]
Pardon me, I was wrongly assuming that this is supported out of the box...
(based on this: https://quasar.dev/vue-components/date#persian-calendar)

**Comment by MilosPaunovic** [comment, marked as answer]
Quasar is using functions from this file to parse Persian dates under the hood:
[https://github.com/quasarframework/quasar/blob/dev/ui/src/utils/private/date-persian.js](https://github.com/quasarframework/quasar/blob/dev/ui/src/utils/private/date-persian.js)

My suggestion (for the easiest solution) would be to add a library which could help you achieve that.

- Example: [https://github.com/jalaali/jalaali-js](https://github.com/jalaali/jalaali-js)

If you think it adds to much overhead to your code, try to look at its source code and write your own version.

**Reply by Shayan-To** [reply]
I went with installing and using `jalaali-js`, as it is what [dev/ui/src/utils/private/date-persian.js](https://github.com/quasarframework/quasar/blob/dev/ui/src/utils/private/date-persian.js) is based on.
